src/validators/suppliers/validator.ts

```diff
@@ -1,11 +1,16 @@
 import { SupplierInterface } from "../../models/supplier";
 import supplierSchema from "./supplierSchema";
+import ValidationError from "../../custom_error/customError";
+import { ValidatorStrategy } from "../validatorStrategy";
 
-const validatorSupplier = (supplier: SupplierInterface): void => {
-  const result = supplierSchema.validate(supplier);
-  if (result.error) {
-    throw `Not a valid supplier: ${result.error.message}`;
+export class SupplierStrategy implements ValidatorStrategy {
```

Comment:
Aici nu ar trebui sa fie o strategie pentru supplier si una pentru product. Ci doua strategii pentru product care fac acelasi lucru dar in moduri diferite